### PR TITLE
Lineage API support

### PR DESCRIPTION
**Pull request by labkey-alan**
#### Rationale
This PR adds support for the lineage API to our API Wrapper, which resolves [Issue 49987](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=49987).

#### Related Pull Requests
* https://github.com/LabKey/labkey-api-r/pull/107

#### Changes
- Add lineage API to experiment module
    - Accessible via API wrappers e.g. `api.experiment.lineage()`

### CODE REVIEW

File: labkey/experiment.py [hunk 1]
```diff
@@ -218,6 +218,55 @@ def save_batches(
...
+    if depth is not None:
+        payload["depth"] = depth
+
+    if parents is not None:
+        payload["parents"] = parents
+
+    if run_protocol_lsid is not None:
+        payload["runProtocolLsid"] = run_protocol_lsid
+
+    return server_context.make_request(lineage_url, payload=payload, method="GET")
```
Comment (cnathe) [inline labkey/experiment.py]: I'm not sure what we support on the server side, but any change the list of lsids array could be long enough where we'd want to do a POST here instead?

File: labkey/experiment.py [hunk 2]
```diff
@@ -218,6 +218,55 @@ def save_batches(
...
+    :param lsids: Array of LSIDs for the seed ExpData, ExpMaterials, or ExpRun
+    :param children: Include children in the lineage response. Defaults to true.
+    :param container_path: labkey container path if not already set in context
+    :param cpas_type: Optional LSID of a SampleSet or DataClass to filter the response. Defaults to include all.
+    :param exp_type: Optional experiment type to filter response -- either "Data", "Material", or "ExperimentRun".
+    Defaults to include all.
+    :param depth: An optional depth argument. Defaults to include all.
+    :param parents: Include parents in the lineage response. Defaults to true.
+    :param run_protocol_lsid: Optional Exp Run Protocol Lsid to filter response. Defaults to include all.
+    """
```
Comment (cnathe) [inline labkey/experiment.py:249]: in the Rlabkey version, I also had 3 boolean params. Were those intentionally left off?
- includeProperties: (boolean) include properties in the lineage response
- includeInputsAndOutputs: (boolean) include inputs and outputs in the lineage response
- includeRunSteps: (boolean) include run steps in the lineage response

**Review by cnathe — APPROVED** [review]

File: labkey/experiment.py (hunk 2 above)
Comment (labkey-alan) [reply labkey/experiment.py:249]: I only took the params we used in our JS client code, but I can add those here easily

Comment (labkey-alan) [reply labkey/experiment.py:249]: Done

File: labkey/experiment.py (hunk 1 above)
Comment (labkey-alan) [reply labkey/experiment.py]: I changed it to POST and it works the same, so I think I'll keep it that way to prevent any potential URL length issues.

**Review by labkey-tchad — COMMENTED** [review]
Should update `setup.cfg` also. Change `description-file` to `description_file` to get rid of a `SetuptoolsDeprecationWarning` and get rid of the `test` alias and the coverage properties

**Pull request merged**